### DISCUSSION TITLE
Obtain configuration dict from server backend and use it in frontend

### DISCUSSION THREAD

**Discussion by DrissiReda** [vercel/next.js#74350] [Help]
### Summary

I am using next.js for my server's frontend and flask python for backend. On server start (runtime) I want my frontend to obtain the dict configuration from the flask backend and keep using it everywhere on my static frontend.
However anything I find on internet involves very complex hooks and modifications of multiple files. Isn't it possible to prepare a variable on server start, export it then use it as variable in all frontend files. Or set it once to localStorage then keep calling that function.

I keep getting errors like :

```
React has detected a change in the order of Hooks called by Navbar. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://react.dev/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useContext                 useContext
2. useState                   useState
3. useEffect                  useEffect
4. undefined                  useContext
```

I'm a beginner and just need the basics, most of the work is done on the backend.

### Additional information

Ideally I'd also like to be able to export everything as static files to be served under nginx for production. Is it possible to still use a json config file that is read at runtime and served by nginx without any node ?
_No response_

### Example

I'll add a minimal example of the code I'm trying:

<details>
<summary>lib/utils.tsx</summary>

```tsx
import { cache } from 'react'

export const revalidate = 60 // revalidate this data every 60 seconds

export const getConfig = cache(async () => {
  const res = await fetch('http://localhost:5000/api/config', { next: { revalidate } })
  if (!res.ok) throw new Error('Failed to fetch config')
  return res.json()
})
```
</details>

<details>
<summary>lib/utils.tsx</summary>

```tsx

```
</details>

<details>
<summary>lib/utils.tsx</summary>

```tsx

```
</details>

<details>
<summary>lib/utils.tsx</summary>

```tsx

```
</details>

<details>
<summary>lib/utils.tsx</summary>

```tsx

```
</details>

**Comment by icyJoseph** [comment]
That order of hooks issue means that you are, somehow doing, for example, but not limited to:

```jsx
const Foo  = () => {
  if (someCondition) {
    useEffect(() => {}, [])
  }
}
```

Show use some code and I can try to help with a plausible solution.

**Reply by DrissiReda** [reply]
thanks @icyJoseph  I added some code to my original post.

what I'm doing currently is hard coded `lib/globals.ts` that is only modified at build time. I'd like to have something that receives config from backend and still build statically to be served by nginx in production in a containerized environment.

Thank you very much !
